src/Product/AttributeMapping/AttributeMappingHelper.php

```diff
@@ -0,0 +1,150 @@
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Condition;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Gender;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\GTIN;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\IsBundle;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Material;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\MPN;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Multipack;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Pattern;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Size;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\SizeSystem;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\SizeType;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\WithMappingInterface;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Helper Class for Attribute Mapping
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Product\AttributeMapping
+ */
+class AttributeMappingHelper implements Service {
+
+
+	private const ATTRIBUTES_AVAILABLE_FOR_MAPPING = [
+		Adult::class,
+		AgeGroup::class,
+		Brand::class,
+		Color::class,
+		Condition::class,
+		Gender::class,
+		GTIN::class,
+		IsBundle::class,
+		Material::class,
+		MPN::class,
+		Multipack::class,
+		Pattern::class,
+		Size::class,
+		SizeSystem::class,
+		SizeType::class,
+	];
+
+	public const CATEGORY_CONDITION_TYPE_ALL    = 'ALL';
+	public const CATEGORY_CONDITION_TYPE_ONLY   = 'ONLY';
+	public const CATEGORY_CONDITION_TYPE_EXCEPT = 'EXCEPT';
+
+	/**
+	 * Gets all the available attributes for mapping
+	 *
+	 * @return array
+	 */
+	public function get_attributes(): array {
+		$destinations = [];
+
+		/**
+		 * @var WithMappingInterface $attribute
+		 */
+		foreach ( self::ATTRIBUTES_AVAILABLE_FOR_MAPPING as $attribute ) {
+			array_push(
+				$destinations,
+				[
+					'id'    => $attribute::get_id(),
+					'label' => $attribute::get_name(),
+					'enum'  => $attribute::is_enum(),
+				]
+			);
+		}
+
+		return $destinations;
+	}
+
+	/**
+	 * Get the attribute class based on attribute ID.
+	 *
+	 * @param string $attribute_id  The attribute ID to get the class
+	 * @return string|null The attribute class path or null if it's not found
+	 */
+	private function get_attribute_by_id( string $attribute_id ): ?string {
+		// Find the attribute class by id using a filter since PHP doesn't support user defined functions for comparison.
+		// array_values is there because array_filter preserves keys index
+		$attribute = array_values(
+			array_filter(
+				self::ATTRIBUTES_AVAILABLE_FOR_MAPPING,
+				function ( $class ) use ( $attribute_id ) {
+					/**
+					 * @var AttributeInterface $class
+					 */
+					return $class::get_id() === $attribute_id;
+				}
+			)
+		);
+
+		return ! empty( $attribute ) ? $attribute[0] : null;
```

Comment:
Since we are only going to return the first match we could simplify this code to make it more readable.

```suggestion
		foreach ( self::ATTRIBUTES_AVAILABLE_FOR_MAPPING as $class ) {
			if ( $class::get_id() === $attribute_id ) {
				return $class;
			}
		}

		return null;
```